dcicutils/env_utils.py

```diff
@@ -691,7 +691,7 @@ def _find_public_url_entry(envname):
 
 
 @if_orchestrated
-def get_env_real_url(envname):
+def get_env_real_url(envname: EnvName) -> UrlString:
```

Comment:
Importantly, this always returns some attempt at naming the server, never `None` or an empty string.